joomla/joomla-cms
# [4.0] Make the tuple length of com_finder configurable

[UPDATE] After the initial discussion, I changed this to make the tuple length of com_finder be configurable. This should make this less controversial. 

## Original text for the old PR
Finder is indexing not only every word, but also every combination of 2- and 3-word-tuples in the text to be indexed. This results in gigantic storage consumption by the index and a performance decrease when indexing of about 20%. At the same time, it only marginally improves the search results. When removing this feature, we are sacrificing the possibility to search for specific phrases of up to 3 terms (like "Joomla is great") for a reduction in necessary storage space of about 2/3. (and the performance increase in indexing)

## How to test
Install 4.0-dev and go to the backend of Smart Search. Enable the Smart Search Plugin and start the indexing process. Notice that it takes a long time (preferably measure the time with a stopwatch). Go to the frontend and open the Smart Search menu item. Search for something and notice the results.
Now apply these changes and go into the backend of Smart Search. Clear the index and rebuild the index, again checking with a stopwatch. You should see a 20% performance increase here. Now go to the frontend and do the same search. You should get the same result.
Do a search for a specific phrase, wrapping the phrase in quotes. See that this does not work anymore.

Please also see #20185 for further improvements for com_finder.

I want to thank [cloudaccess.net](https://www.cloudaccess.net/) for supporting these changes. Without their generous help, this wouldn't be possible.
